src/content/blog/2024/experimental-cleanup-with-nf-boost.md

```diff
@@ -0,0 +1,77 @@
+
+<!-- end-archive-description -->
+
+Automatic cleanup is currently the [oldest open issue](https://github.com/nextflow-io/nextflow/issues/452) on the Nextflow repository. For many users, the ability to quickly delete intermediate files makes the difference between a run being possible or impossible. [Stephen Ficklin](https://github.com/spficklin), the creator of GEMmaker, came up with a clever way to delete intermediate files and even “trick” Nextflow into skipping deleted tasks on a resumed run, which you can read about in the GitHub issue. It involved wiring the intermediate output channels to a “cleanup” process, along with a “done” signal from the relevant downstream processes to ensure that the intermediates were deleted at the right time.
+
+This hack worked, but it required a lot of manual effort to wire up the cleanup process correctly, and it left me wondering whether it could be done automatically. Nextflow should be able to analyze the DAG, figure out when an output file can be deleted, and then delete it! During my time on the Nextflow team, I have implemented this exact idea in a [pull request](https://github.com/nextflow-io/nextflow/pull/3849), but there are still a few challenges to resolve, such as resuming from deleted runs (which is not as impossible as it sounds).
+
+### Introducing nf-boost: experimental features for Nextflow
+
+Many users have told me that they would gladly take the cleanup without the resume, so I found a way to provide the cleanup functionality in a plugin, which I call [nf-boost](https://github.com/bentsherman/nf-boost). This plugin is not just about automatic cleanup – it contains a variety of experimental features, like new operators and functions, that anyone can try today with a few extra lines of config, which is much less tedious than building Nextflow from a pull request. Not every new feature can be implemented via plugin, but for those features that can, it’s nice for the community to be able to try it out before we make it official.
+
+The nf-boost plugin requires Nextflow v23.10.0 or later. You can enable the experimental cleanup by adding the following lines to your config file:
+
+```groovy
+plugins {
+  id 'nf-boost'
+}
+
+boost {
+  cleanup = true
+}
+```
+
+### Automatic cleanup: how it works
+
+The strategy of automatic cleanup is simple:
+
+1. As soon as an output file can be deleted, delete it
+2. An output file can be deleted when:
+  - all downstream tasks that use the output file as an input have completed; AND
+  - the output file has been published (if it needs to be published)
```

Comment:
The linter doesn't seem to like this nested bulleted list. Also it isn't rendered correctly in the preview. If you can't get the nested list to work, I would just collapse it like this:
```md
2. An output file can be deleted when (1) all downstream tasks that use the output file as an input have completed AND (2) the output file has been published (if it needs to be published)
```

Reply:
Yeah, I think that's the best way to do it. Listing won't stop complaining.